tools/wltests/android/create_aboot_img.sh

```diff
@@ -0,0 +1,114 @@
+
+if [ ! -f ${KERNEL} ] ; then
+	c_error "KERNEL image not found: ${KERNEL}"
+	exit $ENOENT
+fi
+
+if [ ! -f ${DTB} ] ; then
+	c_error "DTB not found: ${DTB}"
+	exit $ENOENT
+fi
+
+if [ ! -f ${RAMDISK} ] ; then
+	c_error "RAMDISK image not found: ${RAMDISK}"
+	c_warning "A valid ramdisk image, which matches the device user-space"
+	c_warning "must be deployed by the user under the required path."
+	c_info "Please refer to the ISTALLATION INSTRUCTIONS"
+	c_info "if you don'tknow how to provide such an image."
+	echo
+	exit $ENOENT
+fi
+
+################################################################################
+# Report configuration
+################################################################################
+echo
+c_info "Generate BOOT image:"
+c_info "   $BOOT_IMAGE"
+c_info "using this configuration :"
+c_info "  KERNEL                 : $KERNEL"
+c_info "  RAMDISK                : $RAMDISK"
+c_info "  CMDLINE                : $CMDLINE"
+c_info "  ANDROID_IMAGE_BASE     : $ANDROID_IMAGE_BASE"
+c_info "  ANDROID_IMAGE_PAGESIZE : $ANDROID_IMAGE_PAGESIZE"
+c_info "  ANDROID_OS_VERSION     : $ANDROID_OS_VERSION"
+c_info "  ANDROID_OS_PATCH_LEVEL : $ANDROID_OS_PATCH_LEVEL"
+
+# Optional arguments
+if [ "${ANDROID_TAGS_OFFSET}" ]; then
+    c_info "- ANDROID_TAGS_OFFSET    : ${ANDROID_TAGS_OFFSET}"
+    ANDROID_TAGS_OFFSET="tagsaddr = ${ANDROID_TAGS_OFFSET}"
+fi
+
+if [ "${ANDROID_KERNEL_OFFSET}" ]; then
+    c_info "- ANDROID_KERNEL_OFFSET  : ${ANDROID_KERNEL_OFFSET}"
+    ANDROID_KERNEL_OFFSET="kerneladdr = ${ANDROID_KERNEL_OFFSET}"
+fi
+
+if [ "${ANDROID_BOOT_SIZE}" ]; then
+    c_info "- ANDROID_BOOT_SIZE: ${ANDROID_BOOT_SIZE}"
+    ANDROID_BOOT_SIZE="bootsize = ${ANDROID_BOOT_SIZE}"
+fi
+
+if [ "${ANDROID_PAGE_SIZE}" ]; then
+    c_info "- ANDROID_PAGE_SIZE: ${ANDROID_PAGE_SIZE}"
+    ANDROID_PAGE_SIZE="pagesize = ${ANDROID_PAGE_SIZE}"
+fi
+
+if [ "${ANDROID_SECOND_SIZE}" ]; then
+    c_info "- ANDROID_SECOND_SIZE: ${ANDROID_SECOND_SIZE}"
+    ANDROID_SECOND_SIZE="secondaddr = ${ANDROID_SECOND_SIZE}"
+fi
+
+if [ "${ANDROID_RAMDISK_OFFSET}" ]; then
+    c_info "- ANDROID_RAMDISK_OFFSET : ${ANDROID_RAMDISK_OFFSET}"
+    ANDROID_RAMDISK_OFFSET="ramdiskaddr = ${ANDROID_RAMDISK_OFFSET}"
+fi
+
+cat > "${ARTIFACTS_PATH}"/bootimg-960.cfg << EOL
+${ANDROID_BOOT_SIZE}
+${ANDROID_PAGE_SIZE}
+${ANDROID_KERNEL_OFFSET}
+${ANDROID_RAMDISK_OFFSET}
+${ANDROID_SECOND_SIZE}
+${ANDROID_TAGS_OFFSET}
+name =
+cmdline = ${CMDLINE}
+EOL
+
+################################################################################
+# Generate BOOT image
+################################################################################
+
+# Ensure the output folder exists
+mkdir -p $(dirname $BOOT_IMAGE) &>/dev/null
+
+set -x
+c_info "- IMAGE_DTB: $IMAGE_DTB"
+IMAGE_DTB="$(dirname $KERNEL)/Image-dtb"
+cat "${KERNEL}" "${DTB}" > "${IMAGE_DTB}"
+abootimg --create "${BOOT_IMAGE}" -k "${IMAGE_DTB}" -r "${RAMDISK}" -f "${ARTIFACTS_PATH}"/bootimg-960.cfg
```

Comment:
This `abootimg` tool should be provided by this patch and be available under `tools/wltests/android`

Have a look at how to use mkbootimg here:
https://github.com/ARM-software/lisa/blob/master/tools/wltests/android/create_boot_img.sh#L75
... on the formatting side as well ;-)

Reply:
abootimg can be installed by Ubuntu apt-get command, mkbooting is not official binary. So I think it's more reasonable to directly use Ubuntu official abootimg, so will add checking and give instructions for abootimg.

Reply:
Yes, it that case it make sense to just split out a warning if the binary is not available with a note on which packages should be installed. Thanks.